File: frontend/mobile/lib/model/corn_field_model.dart
class CornField {
  final String id;
  final String fieldName;
  final String location;
  final String cornVariety;
  final String soilType;
  final String growthStage;
  final DateTime plantingDate;
  final String? userId;

  CornField({
    required this.id,
    required this.fieldName,
    required this.location,
    required this.cornVariety,
    required this.soilType,
    required this.growthStage,
    required this.plantingDate,
    this.userId,
  });

  factory CornField.fromJson(Map<String, dynamic> json) {
    return CornField(
      id: json['_id'] ?? json['id'] ?? '',
      fieldName: json['fieldName'] ?? '',
      location: json['location'] ?? '',
      cornVariety: json['cornVariety'] ?? '',
      soilType: json['soilType'] ?? '',
      growthStage: json['growthStage'] ?? 'VE',  // Make sure this is getting the right property
      plantingDate: json['plantingDate'] != null 
          ? DateTime.parse(json['plantingDate']) 
          : DateTime.now(),
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fieldName': fieldName,
      'location': location,
      'cornVariety': cornVariety,
      'soilType': soilType,
      'growthStage': growthStage,
      'plantingDate': plantingDate.toIso8601String(),
      'userId': userId,
    };
  }
}

File: backend/models/cornField.model.js
import mongoose from 'mongoose';
const { Schema } = mongoose;

const cornFieldSchema = new Schema({
  userId: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  fieldName: {
    type: String,
    required: true
  },
  location: {
    type: String,
    required: true
  },
  soilType: {
    type: String,
    required: true
  },
  cornVariety: {
    type: String,
    required: true
  },
  plantingDate: {
    type: Date,
    required: true
  },
  growthStage: {
    type: String,
    required: true,
    default: 'VE' // Initial growth stage
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
}, { timestamps: true });

const CornField = mongoose.model('CornField', cornFieldSchema);

export default CornField;